arch/riscv/lib/memcpy.S

```diff
@@ -89,6 +89,21 @@ ENTRY(memcpy)
 	/* Handle trailing misalignment */
 	beqz a2, 6f
 	add a3, a1, a2
+
+	/* Use word-oriented copy if co-aligned to word boundary */
+	or a5, a1, t6
+	or a5, a5, a3
+	andi a5, a5, 3
+	bnez a5, 5f
```

Comment:
My mistake - The  checking a3 is guarding this.

So it looks very good.
Did you still want to remove the check for less than 128 bytes?

This could be further optimized to allow for word copy with trailing misalignment.
The worst case of trailing misalignment is also 127 bytes, the same as the initial check.
Is it worth the effort?